### PR TITLE
add a method to the session that blocks until it is closed

### LINKED ISSUE #632: implement some way to tell if a session is closed
An application might want to know if a session is closed (e.g. in order to garbage-collect it). Our current `Session` API doesn't support that.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #632.

### CODE REVIEW

File: interface.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ type Session interface {
 	RemoteAddr() net.Addr
 	// Close closes the connection. The error will be sent to the remote peer in a CONNECTION_CLOSE frame. An error value of nil is allowed and will cause a normal PeerGoingAway to be sent.
 	Close(error) error
+	// WaitUntilClosed() blocks until the session is closed.
```
Comment (lucas-clemente) [inline interface.go]: Please add a comment that this signature might change (e.g. we might want to add an error down the road).

File: session_test.go [hunk 2]
```diff
@@ -772,6 +772,17 @@ var _ = Describe("Session", func() {
 			Expect(mconn.written[0][0] & 0x02).ToNot(BeZero()) // Public Reset
 			Expect(sess.runClosed).To(BeClosed())
 		})
+
+		It("unblocks WaitUntilClosed when the run loop exists", func() {
```
Comment (lucas-clemente) [inline session_test.go]: Please use an atomic bool.

**Comment by lucas-clemente** [comment]
Also, you have to update the mocks.

File: session_test.go (hunk 2 above)
Comment (marten-seemann) [inline session_test.go]: We're making our tests race-free? That'll be a long way to go and a lot of bools to change.

**Pull request merged**